truckersmp_cli/steamcmd.py

```diff
@@ -20,6 +20,39 @@
 from .variables import AppId, Args, Dir, URL
 
 
+def run_steamcmd(cmdline_seq, env=None):
+    """
+    Run SteamCMD using given command line.
+
+    When it fails, sys.exit() is called and the program exits abnormally.
+
+    cmdline_seq: SteamCMD command line (list or tuple)
+    env: "env" argument for subprocess.Popen
+    """
+    env_str = ""
+    if env is not None:
+        env_vars = ("WINEDEBUG", "WINEARCH", "WINEPREFIX", "WINEDLLOVERRIDES")
+        for i, var in enumerate(env_vars):
+            if i != 0:
+                env_str += "\n  "
+            env_str += "{}={}".format(var, env[var])
+        env_str += "\n  "
+    cmd_str = ""
+    for i, arg in enumerate(cmdline_seq):
+        if arg.startswith("+"):
+            # add newline before SteamCMD commands (e.g. "+login")
+            cmd_str += "\n    "
+        elif i != 0:
+            cmd_str += " "
+        cmd_str += arg
+    logging.info("Running SteamCMD:\n  %s%s", env_str, cmd_str)
+
+    try:
+        subproc.check_call(cmdline_seq, env=env)
+    except subproc.CalledProcessError:
+        sys.exit("SteamCMD exited abnormally")
```

Comment:
This looks a bit strange is there a reason you're assembling the commandline beforehand and giving it as a whole instead of assembling it in the function itself? I thought of something (imho) simple like (quick hack, not verified):

~~~ python
def run_steamcmd(steamcmd, dir, appid, branch="public", platform=None, env=None):
    cmd = [
        steamcmd,
        if platform is not None: "+@sSteamCmdForcePlatformType", "windows",
        "+login", Args.account,
        "+force_install_dir", dir,
        "+app_update", appid,
        "-beta", branch,
        "validate",
        "+quit",
    ]
~~~